Assets/HotmaxEditorVR/loadPropsMenu.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+public class loadPropsMenu : MonoBehaviour
+{
+    //--local refs
+    private stateManager _stateManagerMutatorRef;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
+    private bool _playerIsLocomoting = stateManager.playerIsLocomoting;
+
+    //--child gameObject refs
+    public GameObject propTiles;
+    
+    //--Lists to handle getting unique props in menu
+    private List<string> propName = new List<string>();
+    private List<GameObject> propObject = new List<GameObject>();
+    private List<List<GameObject>> chunkedList = new List<List<GameObject>>();
+
+    //--if false, disabled initial props on stage
+    //--TODO: remove when props are received from PropClass
+    public bool propIsInStageAlready = false;
```

Comment:
I think this is one is veering a little too far into demo-land for a master branch pull, now that i'm reading this again.
Right now it seems like props have to be active in the scene for this class to work, which means that the menu has to be on immediately and then turn off the props immediately after that.. which is a lot of untying we'll have to do later. 

I would just make a very simple `PropsController` singleton class attached to props gameobject and stub out a public list of gameobjects within that class (maybe on start it just reads whats in it, adds it to the list if distinct, and deactivates them).
That way this class isn't dependent on a transform tree lookup - it's just getting a public list on start, regardless if the objects are active or not. That way this class has a consistent API and the stub code gets moved to `PropsController` where it should be.

then i would remove the propIsInStageAlready stuff, and loop through the public List instead of transform tree. Also don't need distinct check here if that is in props stub class already.

Reply:
got it